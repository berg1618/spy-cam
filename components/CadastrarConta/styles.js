import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
    b: {
        height: '100%',
        backgroundColor: '#413C45',
    },
    scroll: {
        width: '100%'
    },
    containerCenter: {
        margin: 'auto',
        alignItems: 'center',
    },
    titleApp: {
        fontSize: 55,
        textAlign: 'center',
        color: '#F2F0F4',
        marginTop: 100,
        fontWeight: '800',
    },
    logo: {
        marginTop: 48,
        marginBottom: 62,
        width: 70,
        height: 70,
    },
    formContainer: {
        width: '80%',
        // height:'40%',
        marginBottom: 20,
        backgroundColor: '#F2F0F4',
        position: 'relative',
        display: 'flex',
        alignItems: 'center',
        borderRadius: 10,
        padding: 15,

    },
    titleContainer: {
        fontSize: 20,
        color: '#413C45',
        fontWeight: '500',
        marginTop: 10,
        marginBottom: 20,
    },
    icon: {
        bottom: 2,
        position: 'relative',
    },
    inputWrapper: {
        flexDirection: 'row',
        alignItems: 'center',
        width: '60%',
    },
    inputText: {
        marginLeft: 8,
        position: 'relative',
        flex: 1,
        borderBottomWidth: 1,
        borderBottomColor: '#000',
        marginBottom: 5,
    },
    entrada: {
        flexDirection: 'row',
        marginBottom: 3,
    },
    botao: {
        textAlign: 'center',
        backgroundColor: '#413C45',
        color: '#F2F0F4',
        borderRadius: 10,
        paddingTop: 6,
        paddingBottom: 6,
        paddingLeft: 45,
        paddingRight: 45,
        marginTop: 35,
        marginBottom: 5,
    },
    mensagem: {
        fontSize: 10,
    },
    textBold: {
        color: '#0057FF',
        fontSize: 9,
    },
})

export default styles