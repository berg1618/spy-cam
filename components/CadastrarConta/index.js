import { Image, Text, View, TextInput, TouchableOpacity, ScrollView, SafeAreaView, Alert } from "react-native";
import styles from './styles.js'
import React, {useState} from "react";

const CadastrarConta = ({navigation}) => {
  const [nome, onChangeNome] = useState("")
  const [email, onChangeEmail] = useState("")
  const [senha, onChangeSenha] = useState("")
  const [confSenha, onChangeConfSenha] = useState("")

  const createTwoButtonAlert = (title, msg) =>
  Alert.alert(title? title: 'Erro de autenticação!', msg, [
    {text: 'OK', onPress: () => console.log('OK Pressed')},
  ]);


  const cadastrarUsuario = async (nome, email, senha, confSenha) => {
    try {
      // exibir alerta caso algum campo esteja vazio
     if (!nome || !email || !senha || !confSenha) {
         return createTwoButtonAlert("", "Preencha todos os campos!")
     }

    // verificar se senha e confirmacao sao iguais
    if (senha != confSenha) {
      return createTwoButtonAlert("Erro!", "as senhas que você digitou não são iguais!")
    }
    
      navigation.navigate("pageInicial");
    }
    catch (err) {
      // ### essa msg pode ser substituida por uma mais personalizada
      createTwoButtonAlert("problemas : -( ", "")
    }
  }

  return (
    // <View style={styles.b}>
      <SafeAreaView style={styles.b}>
        <ScrollView style={styles.scroll}>
          <View style={styles.containerCenter}>
            <View>
            <Text style={styles.titleApp}>
              Spy Cam
            </Text>
          </View>

          <Image
            style={styles.logo}
            source={require('../../assets/iconReconhecimento.png')}
            />

            <View style={styles.formContainer}>

              <Text style={styles.titleContainer}>Faça seu cadastro</Text>
              <View style={styles.entrada}>
                <Image
                  style={styles.icon}
                  source={require('../../assets/iconNome.png')} 
                />
                <TextInput
                  style={styles.inputText}
                  placeholder="Nome"
                  keyboardType="default"
                  onChangeText={onChangeNome}
                  value={nome}
                />
              </View>
              <View> 
              <Text>____________________________</Text>
              </View>


              <View style={styles.entrada}>
              <Image
                style={styles.icon} 
                source={require('../../assets/iconEnail.png')}
              />
                <TextInput
                  style={styles.inputText}
                  placeholder="Email"
                  keyboardType="email-address"
                  onChangeText={onChangeEmail}
                  value={email}
                />
              </View>
              <View> 
              <Text>____________________________</Text>
              </View>
            
              <View style={styles.entrada}>
                <Image
                  style={styles.icon}
                  source={require('../../assets/iconPassword.png')}
                />
                <TextInput
                  style={styles.inputText}
                  placeholder="Senha"
                  keyboardType="default"
                  onChangeText={onChangeSenha}
                  value={senha}
                />
              </View>
              <View> 
              <Text>____________________________</Text>
              </View>

              <View style={styles.entrada}>
                <Image
                  style={styles.icon}
                  source={require('../../assets/iconPassword.png')}
                />
                <TextInput
                  style={styles.inputText}
                  placeholder=" Confirme sua senha"
                  keyboardType="default"
                  onChangeText={onChangeConfSenha}
                  value={confSenha}
                />
              </View>
              <View> 
              <Text>____________________________</Text>
              </View>

              <TouchableOpacity
                onPress={() => cadastrarUsuario(nome, email, senha, confSenha)}
              >
                  <Text style={styles.botao}>Cadastre-se</Text>
              </TouchableOpacity>

              <Text
                onPress={() => {
                navigation.navigate("login")
              }}
              style={styles.mensagem}>
                Ja possui conta? <Text style={styles.textBold}>Login</Text>
              </Text>
            
            </View>
          </View>
        </ScrollView>
      </SafeAreaView>
    // </View>
  );
}

export default CadastrarConta;