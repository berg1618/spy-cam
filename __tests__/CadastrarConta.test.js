import React from 'react';
import renderer from 'react-test-renderer';
import CadastrarConta from '../components/CadastrarConta';

test('testando o componente de cadastrar conta', () => {
    const tree = renderer.create(<CadastrarConta />).toJSON();
    expect(tree).toMatchSnapshot();
  });