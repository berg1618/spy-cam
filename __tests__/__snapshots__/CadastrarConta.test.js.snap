// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testando o componente de cadastrar conta 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#413C45",
      "height": "100%",
      "margin": "auto",
    }
  }
>
  <View>
    <Text
      style={
        {
          "color": "#F2F0F4",
          "fontSize": 55,
          "fontWeight": "800",
          "marginTop": 100,
          "textAlign": "center",
        }
      }
    >
      Spy Cam
    </Text>
  </View>
  <Image
    source={
      {
        "testUri": "../../../assets/iconReconhecimento.png",
      }
    }
    style={
      {
        "height": 70,
        "marginBottom": 62,
        "marginTop": 48,
        "width": 70,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F2F0F4",
        "borderRadius": 10,
        "display": "flex",
        "height": "40%",
        "padding": 15,
        "position": "relative",
        "width": "90%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#413C45",
          "fontSize": 20,
          "fontWeight": "500",
          "marginBottom": 20,
          "marginTop": 10,
        }
      }
    >
      Fa√ßa seu cadastro
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": -12,
          "marginTop": 5,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/iconNome.png",
          }
        }
        style={
          {
            "left": 2,
            "position": "relative",
            "top": 5,
          }
        }
      />
      <TextInput
        keyboardType="default"
        placeholder="Nome"
        style={
          {
            "marginLeft": 8,
            "position": "relative",
            "width": 150,
          }
        }
      />
    </View>
    <View>
      <Text>
        ____________________________
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": -12,
          "marginTop": 5,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/iconEnail.png",
          }
        }
        style={
          {
            "left": 2,
            "position": "relative",
            "top": 5,
          }
        }
      />
      <TextInput
        keyboardType="email-address"
        placeholder="Email"
        style={
          {
            "marginLeft": 8,
            "position": "relative",
            "width": 150,
          }
        }
      />
    </View>
    <View>
      <Text>
        ____________________________
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": -12,
          "marginTop": 5,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/iconPassword.png",
          }
        }
        style={
          {
            "left": 2,
            "position": "relative",
            "top": 5,
          }
        }
      />
      <TextInput
        keyboardType="default"
        placeholder="Senha"
        style={
          {
            "marginLeft": 8,
            "position": "relative",
            "width": 150,
          }
        }
      />
    </View>
    <View>
      <Text>
        ____________________________
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": -12,
          "marginTop": 5,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/iconPassword.png",
          }
        }
        style={
          {
            "left": 2,
            "position": "relative",
            "top": 5,
          }
        }
      />
      <TextInput
        keyboardType="default"
        placeholder=" Confirme sua senha"
        style={
          {
            "marginLeft": 8,
            "position": "relative",
            "width": 150,
          }
        }
      />
    </View>
    <View>
      <Text>
        ____________________________
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "backgroundColor": "#413C45",
            "borderRadius": 10,
            "color": "#F2F0F4",
            "marginBottom": 5,
            "marginTop": 35,
            "paddingBottom": 6,
            "paddingLeft": 45,
            "paddingRight": 45,
            "paddingTop": 6,
            "textAlign": "center",
          }
        }
      >
        Cadastre-se
      </Text>
    </View>
    <Text
      onPress={[Function]}
      style={
        {
          "fontSize": 10,
        }
      }
    >
      Ja possui conta? 
      <Text
        style={
          {
            "color": "#0057FF",
            "fontSize": 9,
          }
        }
      >
        Login
      </Text>
    </Text>
  </View>
</View>
`;
